### DISCUSSION TITLE
Oracle Connection using oracledb

### DISCUSSION THREAD

**Discussion by dkowalcz** [oracle/python-oracledb#90] [Q&A]
I have code using cx_oracle with the following connection string:

    # Establish the database connection
    v_service = "xxsod123"
    connection = cx_Oracle.connect(v_uname, v_pword, v_service)

When I convert over to the new python-oracledb this connection using a serivce alias name no longer works. I have to code the name listed in the listener file on the server :

    v_dsn = "server-scan.company.com:1599/xxsod202"
    connection = oracledb.connect(user=v_uname, password=v_pword, dsn=v_dsn)
          or
    v_params = oracledb.ConnectParams(host="server-scan.company.com", port=1599 , service_name="xxsod202")
    connection = oracledb.connect(user=v_uname, password=v_pword, params=v_params)

Is there a way I can just code the service alias name like I did in cx_oracle?

**Comment by anthony-tuininga** [comment]
Can you share the relevant parts of the tnsnames.ora file? I believe this has been noted before but the last time I tried to figure out the issue I was not able to reproduce. I am assuming that if you enable thick mode (by adding a call to `oracledb.init_oracle_client()`) that the code works as before?

**Comment by dkowalcz** [comment]
I tried that - we use OID for connections and do not have a tnsnames file.

**Reply by anthony-tuininga** [reply]
So did it work? What error did you get? Thin mode does not support OID (aka LDAP) so you will have to use thick mode if you want to continue using OID.

**Reply by dkowalcz** [reply]
I added the following to my code and it now works: 

v_lib = os.environ.get("ORACLE_HOME") + "/bin"
oracledb.init_oracle_client(lib_dir=v_lib)

**Reply by anthony-tuininga** [reply]
If you didn't need that with cx_Oracle, you shouldn't need that with oracledb either. The two work identically with respect to the Oracle Client libraries! If you're happy, though, you can close this discussion.
